src/main/java/org/commcare/core/network/CommCareNetworkServiceGenerator.java

```diff
@@ -39,13 +49,40 @@
         return response;
     };
 
+    private static Interceptor driftInterceptor = new Interceptor() {
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+            Request request = chain.request();
+            Response response = chain.proceed(request);
+            ICommCarePreferenceManager commCarePreferenceManager = CommCarePreferenceManagerFactory.getCommCarePreferenceManager();
+            if (commCarePreferenceManager != null) {
```

Comment:
The way `CommCarePreferenceManagerFactory` is implemented right now, anyone who ever wants to use it will have to first do this check for if `CommCarePreferenceManagerFactory.getCommCarePreferenceManager()` is null. Would it maybe make more sense to have the default state of the factory be to have a dummy implementation of `ICommCarePreferenceManager` always initialized?

Reply:
I am a bit conflicted in this. I will surely love not to do these null checks though adding a dummy class would just create more complexity in code base.

Reply:
Well what if instead of having the `ICommCarePreferenceManager` interface, you just turn that into a class, and then `AndroidPreferenceManager` can extend it?

Reply:
I think that will result in PreferenceManager classes not to bind to any contract which can get messy since we might add a particular method in one preference manager class and not in other. So I would actually prefer adding a dummy class to removing the interface if we do wanna avoid null checks for this.

Reply:
I think we should add the dummy class then. Otherwise it's going to be very non-obvious to later users of this interface that the pref manager might be null, and could lead to unnecessary NPEs in the code.

Reply:
I feel like that will cause more checks in the code. For example lets say we decide to return -1 from `getLong` method of the `DummyCommCarePreferenceManager` class. Then the calling method will never know if it's a valid value or not and will again need to some sort of checks to determine this.  

Also today AS gives you a warning if you don't do a null check before accessing `ICommCarePreferenceManager` because of the `Nullable` annotation on `CommCarePreferenceManagerFactory::getCommCarePreferenceManager()`

Reply:
Hmm okay that's fair, I'll concede :)